radix-engine-interface/src/blueprints/resource/fungible/fungible_bucket.rs

```diff
@@ -7,27 +7,45 @@ pub const FUNGIBLE_BUCKET_BLUEPRINT: &str = "FungibleBucket";
 
 pub const FUNGIBLE_BUCKET_LOCK_AMOUNT_IDENT: &str = "lock_amount";
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor, ManifestSbor)]
 pub struct FungibleBucketLockAmountInput {
     pub amount: Decimal,
 }
 
+pub type FungibleBucketLockAmountManifestInput = FungibleBucketLockAmountInput;
+
 pub type FungibleBucketLockAmountOutput = ();
 
 pub const FUNGIBLE_BUCKET_UNLOCK_AMOUNT_IDENT: &str = "unlock_amount";
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor, ManifestSbor)]
 pub struct FungibleBucketUnlockAmountInput {
     pub amount: Decimal,
 }
 
+pub type FungibleBucketUnlockAmountManifestInput = FungibleBucketUnlockAmountInput;
+
 pub type FungibleBucketUnlockAmountOutput = ();
 
 pub const FUNGIBLE_BUCKET_CREATE_PROOF_OF_AMOUNT_IDENT: &str = "create_proof_of_amount";
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor, ManifestSbor)]
 pub struct FungibleBucketCreateProofOfAmountInput {
     pub amount: Decimal,
 }
 
+pub type FungibleBucketCreateProofOfAmountManifestInput = FungibleBucketCreateProofOfAmountInput;
+
 pub type FungibleBucketCreateProofOfAmountOutput = Proof;
+
+pub type FungibleBucketPutInput = BucketPutInput;
```

Comment:
Maybe. What I think is a better idea in my opinion, and it's something that I wanted to do for this PR but didn't quite have the time, is to create a new concept of an invocation in the `radix-engine-interface` crate. This would be a trait that looks like the following:

```rust
pub trait Invocation {
    const IDENT: &'static str;
    const EXPORT_NAME: &'static str;

    type ScryptoInput: ScryptoEncode + ScryptoDecode + ScryptoCategorize + ScryptoDescribe;
    type ManifestInput: ManifestEncode + ManifestDecode + ManifestCategorize + ScryptoDescribe;

    type ScryptoOutput: ScryptoEncode + ScryptoDecode + ScryptoCategorize + ScryptoDescribe;
    type ManifestOutput: ManifestEncode + ManifestDecode + ManifestCategorize + ScryptoDescribe;
}
```

We could then define invocation types for each of the invocations that we have:

```rust
pub struct FungibleBucketLockAmountInvocation;
impl Invocation for FungibleBucketLockAmountInvocation {
    const BLUEPRINT: &'static str = FUNGIBLE_BUCKET_BLUEPRINT;
    const IDENT: &'static str = FUNGIBLE_BUCKET_LOCK_AMOUNT_IDENT;
    const EXPORT_NAME: &'static str = FUNGIBLE_BUCKET_LOCK_AMOUNT_IDENT;

    type ScryptoInput = FungibleBucketLockAmountInput;
    type ManifestInput = FungibleBucketLockAmountInput;

    type ScryptoOutput = FungibleBucketLockAmountOutput;
    type ManifestOutput = FungibleBucketLockAmountOutput;
}
```

What I like about these invocation types is that they collect a couple of important constants and types into one place and perhaps even _force_ us to define scrypto and manifest input and output types. I've found this concept of an invocation or invocation interface to be very useful and very much needed in many places, the latest example is the typed invocations where having this would've been very nice.

Reply:
Yes, I agree, have been thinking this for a while.

We actually used to have something like that a couple of years ago but I think someone got rid of it 😅. But I think it’s useful in a “stores type metadata” kind of way.

Reply:
It might look something like:

```rust
pub trait NativeMethodInvocation: NativeInvocation {
    const PACKAGE_ADDRESS: PackageAddress;
    const BLUEPRINT_NAME: &'static str;
    const IDENT: &'static str;
    const RECEIVER: ReceiverInfo;
}

pub trait NativeFunctionInvocation: NativeInvocation {
    const PACKAGE_ADDRESS: PackageAddress;
    const BLUEPRINT_NAME: &'static str;
    const IDENT: &'static str;
}

pub trait NativeInvocation {
    const EXPORT_NAME: &'static str;

    type ScryptoInput: ScryptoEncode + ScryptoDecode + ScryptoCategorize + ScryptoDescribe;
    type ManifestInput: ManifestEncode + ManifestDecode + ManifestCategorize + ScryptoDescribe;

    type ScryptoOutput: ScryptoEncode + ScryptoDecode + ScryptoCategorize + ScryptoDescribe;
    type ManifestOutput: ManifestEncode + ManifestDecode + ManifestCategorize + ScryptoDescribe;
}
```